apps/antalmanac/src/stores/AppStore.ts

```diff
@@ -41,11 +41,13 @@ class AppStore extends EventEmitter {
             return theme === null ? 'auto' : theme;
         })();
 
-        window.addEventListener('beforeunload', (event) => {
-            if (this.unsavedChanges) {
-                event.returnValue = `Are you sure you want to leave? You have unsaved changes!`;
-            }
-        });
+        if (typeof window !== 'undefined') {
```

Comment:
Vitest runs in Node, not the browser, so do this check so it doesn't break. Maybe TODO is to find a way to get Vitest to pretend it's running in the browser.